File: 프로그래머스/2/42578. 의상/의상.java
/*
종류별로 최대 1가지
한개이상의 의상
코니가 가진 의상들이 담긴 2차원 배열 clothes
서로 다른 옷의 조합의 수를 return 
*/

/*
1. 종류별로 몇개인지 DICTIONARY로 저장
2. (각 종류 +1) 다 곱하고 -1*/

import java.util.*;
import java.lang.Math;
import java.io.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> types = new HashMap<>();
        for(String[] cloth: clothes){
            if (types.containsKey(cloth[1])){
                types.put(cloth[1],types.get(cloth[1]) +1);
            }
            else{
                 types.put(cloth[1],1);
            }
            
        }
        for(String type: types.keySet()){
            answer *= (types.get(type)+1);
        }
        return (answer-1);
    }
}

File: 프로그래머스/2/42578. 의상/의상.py
"""
- 서로 다른 옷의 조합 수 구하기
- 종류별로 최대 1가지 의상만 가능(안 입는 것도 가능)
- 옷을 하나는 입어야 됨

=>
각 type별 옷의 종류를 각각 구하기 [dict]
(각 type별 옷의 종류 + 1)끼리 곱하고 - 1(<- 아무것도 안 입는 경우)
"""
def solution(clothes):
    answer = 1
    
    # 각 type별 옷의 종류 가짓수 dict
    type = dict()
    
    for cloth in clothes:
        name = cloth[1]
        if type.get(name):
            type[name] += 1
        else:
            type[name] =1

    # 이들끼리 곱하기
    for i in type.values():
        answer *= (i+1)
    
    return answer-1